### ISSUE TITLE
Double click Audio file in the FileSystem to start playback immediately.

### ISSUE THREAD

**Issue by golddotasksquestions** [godotengine/godot-proposals#556]
**Describe the project you are working on:** 
Implementing sounds.

**Describe the problem or limitation you are having in your project:**
Often it's not enough to give soundFX descriptive names. I have to listen to them to make sure it's the one I want to use. This is very tedious right now, as you have to double click the file in the FileSystem, then go over to the Inspector and press the play button.
[Because of this issue](https://github.com/godotengine/godot/issues/36816) it's even more tedious when you feel like you need to listen to the sound more than once, which is very common.
![playback_audiofile2](https://user-images.githubusercontent.com/47016402/75983996-e7f1dd00-5ee9-11ea-9d5f-3552955b661b.gif)

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
Simply double clicking the file in the FileSystem should immediately start the playback. Double clicking the file while it is in playback, should immediately start the playback again from the beginning.
![playback_audiofile_doubleclick](https://user-images.githubusercontent.com/47016402/75984229-49b24700-5eea-11ea-9f59-8eddda5695bf.gif)

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
Unfortunately I can't provide any code.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
Would be used often by anyone who works with more than a hand full of audio files.

**Is there a reason why this should be core and not an add-on in the asset library?:**
I think it's a necessary improvement to usability.

**Comment by akien-mga** [comment]
I'm not sure it's a good idea. Double-clicking is necessary to load the AudioStreamSample in the inspector, which can be done to then listen to it indeed, but also simply to review or change its properties (format, loop, etc.). I'm not sure it's a good idea to force making this action play a sound when you might not actually want to listen to the sample.

**Comment by golddotasksquestions** [comment]
@akien-mga Can you give me use case when you would _**not**_ want to play that sound on a double click?

**Comment by akien-mga** [comment]
> Can you give me use case when you would not want to play that sound on a double click? I can't think of one.

As I mentioned, if you want to review or edit flags without actually playing the sound. Double click is the way to edit a resource, and playing it is not necessarily what users associate with editing.

> Even if there is one, I feel like the cases I want to have sound played on a double click in the FileSystem by far outnumber those in which I might not want that to happen.

That might be true, that's why I wrote "I'm not sure" twice. I'm raising a potential concern so that it can be discussed and either confirmed by some or discarded as not an actual problem.

**Comment by golddotasksquestions** [comment]
@akien-mga 
I think it's good you raised your concern! That's what these proposals are for, right?

> As I mentioned, if you want to review or edit flags without actually playing the sound. 
That's certainly a valid usecase. However, it seems to me an incredibly rare one at that. How often do you edit your resource flags after import? 99.9% of all times I would want those to be set either before import or in the import settings.
For those who do happen to do this a lot for whatever reason though, what would you think about a checkbox in the Project Settings to disable immediate playback on double click?

**Comment by Xrayez** [comment]
I agree that assigning double-click to act as "Play" for sound samples might not always be the expected behavior.

What about having some tool button next to sound samples? Just like the scene tree has buttons for nodes:

![godot-fs-sound-play-icons](https://user-images.githubusercontent.com/17108460/75994932-5c397a00-5f04-11ea-9b33-f795799c94f3.png)

You could go further and implement more buttons for other resources if this kind of thing is implemented on the FileSystemDock level.

**Comment by golddotasksquestions** [comment]
I would find it more intuitive to double click on the name. A symbol like that is much harder to hit. Also if the FileSystem panel is too wide, you would have to readjust it first, so you know which icon belongs to which soundfile.
One solution could be to make the icon much wider, more like a button and align right at the end of the file name. But if you then have very varying filename lengths, those buttons are again not aligned underneath each other, defeating the purpose to quickly test-hear the files for comparison.

So again, I'd think a simple double click with a "disable" checkbox in the Project Settings would be the cleanest solution.

@Xrayez I'm curious, are you thinking of another usecase, other than the one Akien already mentioned?

**Comment by Xrayez** [comment]
@golddotasksquestions comment from https://github.com/godotengine/godot/pull/36819#issuecomment-595242814:

> Won't this make the sound play when clicking an AudioStream resource in the Inspector? This is probably too invasive, especially for music.

For short samples the proposed solution is acceptable, but for longer samples such as music you'd have to do the opposite: turning off the music every time the file is double-clicked.

Making this optional is the way to go I think. I'd personally enable this feature while working solo vs being in public where I tend to mute sounds to avoid unnecessary attention. So my opinion is biased. 😛

**Comment by groud** [comment]
Instead of a double-click (which is usually used to edit something, not to play it, in Godot), I think a simple autoplay mode that can be enabled in the inspector (next to the play button) would be enough. It would automatically play the sound when you select one. This is what Ableton Live does and it's quite practical.

**Comment by golddotasksquestions** [comment]
@Xrayez  If I don't have headphones plugged in, in public, I just mute my laptop stuck_out_tongue

**Comment by IronicallySerious** [comment]
The UI in Unreal gives a separate button on the music file icon to play it. It seems to work for most people so putting a small play button  on the right corner of a music file item in the FileSystem tree might be a solution.

As per the discussion above, having an option to disable this should also be there IMO because one of the key features in the 3.2 editor was adding editor feature disabling from a menu, and it would help the seasoned Godot game devs.

**Comment by Gianclgar** [comment]
In audio and producer world the key to start/stop playback use to be **spacebar**. Also on YouTube and a lot of audio/video player implementations. Don't know if it conflicts with anything else but it would be pretty straightforward.

**Comment by Calinou** [comment]
Pressing <kbd>Space</kbd> to play the sound in the AudioStreamPlayer resource preview has been implemented in https://github.com/godotengine/godot/pull/48378.
